x/ccv/provider/keeper/hooks.go

```diff
@@ -64,8 +64,43 @@ func (h Hooks) AfterUnbondingInitiated(ctx sdk.Context, ID uint64) error {
 	return nil
 }
 
-// Define unimplemented methods to satisfy the StakingHooks contract
+func ValidatorConsensusKeyInUse(k *Keeper, ctx sdk.Context, valAddr sdk.ValAddress) bool {
```

Comment:
Why the need of an extra function instead of adding the logic directly in the `AfterValidatorCreated` hook? For UTs, we can use the hook directly?

Reply:
What would be the benefit of inlining into AfterValidatorCreated? I think that would make it harder to understand the functionality and it would be less self documenting.

The status quo in writing software is *not* to inline everything AFAIU.

Reply:
Then move it to `key_assignment.go` and the test to `key_assignment_test.go`.